### DISCUSSION TITLE
windows support

### DISCUSSION THREAD

**Discussion by ihebmeftah** [serverpod/serverpod#1021] [Q&A] [answered]
Why every time when I change yaml file or create another endpoint, I need to run /setup-table in Windows . (Not like Macos in the tutorials )

**Comment by yahu1031** [comment]
Really? Can you please share the flutter and serverpod versions with us once? @ihebmeftah

**Reply by ihebmeftah** [reply]
[√] Flutter (Channel stable, 3.7.12, on Microsoft Windows [version 10.0.22621.1848], locale fr-FR)
    • Flutter version 3.7.12 on channel stable at C:\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 4d9e56e694 (9 weeks ago), 2023-04-17 21:47:46 -0400
    • Engine revision 1a65d409c7
    • Dart version 2.19.6
    • DevTools version 2.20.1

[X] Windows Version (Unable to confirm if installed Windows version is 10 or greater)

[√] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
    • Android SDK at C:\Users\ihebm\AppData\Local\Android\Sdk
    • Platform android-33, build-tools 33.0.2
    • ANDROID_HOME = C:\Users\ihebm\AppData\Local\Android\Sdk
    • Java binary at: C:\Program Files\Android\Android Studio\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-b2043.56-9586694)
    • All Android licenses accepted.

[X] Chrome - develop for the web (Cannot find Chrome executable at .\Google\Chrome\Application\chrome.exe)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[X] Visual Studio - develop for Windows
    X Visual Studio not installed; this is necessary for Windows development.
      Download at https://visualstudio.microsoft.com/downloads/.
      Please install the "Desktop development with C++" workload, including all of its default components

[√] Android Studio (version 2022.2)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-b2043.56-9586694)

[√] VS Code (version 1.79.2)
    • VS Code at C:\Users\ihebm\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.66.0

[√] Connected device (3 available)
    • sdk gphone64 x86 64 (mobile) • emulator-5554 • android-x64    • Android 13 (API 33) (emulator)
    • Windows (desktop)            • windows       • windows-x64    • Microsoft Windows [version 10.0.22621.1848]
    • Edge (web)                   • edge          • web-javascript • Microsoft Edge 114.0.1823.43

[√] HTTP Host Availability
    • All required HTTP hosts are available

**Reply by ihebmeftah** [reply]
i used the last version in serverpod

**Reply by ihebmeftah** [reply]
Package serverpod_cli is currently active at version 1.1.1.
The package serverpod_cli is already activated at newest available version.
To recompile executables, first run `dart pub global deactivate serverpod_cli`.
Installed executable serverpod.
Activated serverpod_cli 1.1.1.

**Reply by yahu1031** [reply]
@ihebmeftah The actual case will be that you need to run the setup-table once, either in mac/windows. Not sure why this happening. And AFAIU your case, This will also give you an error that says can't create a container cause it already exists when you run the `setup-tables.cmd` script. In case you change anything in the `*.yaml` file(models), Please consider the generate command. So that it will create a `tables.pgsql` script and you need to run the script in the DB to create/Alter the tables, But not `setup-tables.cmd`. Anyhow, I will try checking this out. 

> ***(PROMOTION: For better and seamless handling of the serverpod projects, Try out [Serverpod VSCode Extension](https://marketplace.visualstudio.com/items?itemName=minnu1031.serverpod))***


BTW @Isakdl @vlidholt Can you please check this PR #528 and try making these changes in the `main`.

**Reply by ihebmeftah** [reply]
for more explanation: 
when I change my yaml file and run serverpod generate, dart bin/main i receive status code 400. To fix it usually, I need to delete the container in docker and run setup-tables docker-compose up, serverpod generate, and dart bin/main. i hope this clear :D

**Reply by yahu1031** [reply]
Let me check this.

**Reply by yahu1031** [reply]
Check #511 @ihebmeftah

**Reply by ihebmeftah** [reply]
same problem

**Reply by ihebmeftah** [reply]
if u want i do screen record !

**Reply by yahu1031** [reply]
@ihebmeftah Can't able to replicate :(

Sys info

```console
Edition	Windows 11 Pro Insider Preview
Version	22H2
Installed on	‎20-‎01-‎2023
OS build	25281.1000
Experience	Windows Feature Experience Pack 1000.25281.1000.0
```

**Reply by yahu1031** [reply, marked as answer]
@ihebmeftah https://github.com/serverpod/serverpod/issues/511#issuecomment-1601633724 This will help you with the `400 status code` issue. Mark it as answer if it works.
